idaes/apps/grid_integration/pricetaker/tests/test_clustering.py

```diff
@@ -0,0 +1,281 @@
+    ):
+        generate_daily_data(raw_data=[1, 2, 3, 4], horizon_length=24)
+
+
+@pytest.mark.unit
+def test_daily_data_logger_message2(caplog):
+    """Tests non-integer-multiple length warning in generate_daily_data function"""
+
+    with caplog.at_level(idaeslog.WARNING):
+        daily_data = generate_daily_data(raw_data=list(range(8)), horizon_length=3)
+        assert (
+            "Length of the price signal is not an integer multiple of horizon_length.\n"
+            "\tIgnoring the last 2 elements in the price signal."
+        ) in caplog.text
+
+    assert isinstance(daily_data, pd.DataFrame)
+    assert daily_data.shape == (2, 3)
+
+
+@pytest.mark.unit
+def test_generate_daily_data():
+    """Tests the generate_daily_data function"""
+
+    raw_data = list(range(9))
+    daily_data = generate_daily_data(raw_data, horizon_length=3)
+    assert daily_data.shape == (3, 3)
+    assert daily_data.equals(
+        pd.DataFrame({1: [0, 1, 2], 2: [3, 4, 5], 3: [6, 7, 8]}).transpose()
+    )
+
+
+@pytest.mark.unit
+@pytest.mark.skipif(
+    not sklearn_avail, reason="sklearn (optional dependency) not available"
+)
+def test_cluster_lmp_data(dummy_data):
+    """Tests the cluster_lmp_data function"""
+
+    data_clusters, weights = cluster_lmp_data(
+        raw_data=dummy_data, horizon_length=2, n_clusters=3, seed=42
+    )
+    assert weights == {1: 4, 2: 4, 3: 4}
+    assert data_clusters == {
+        1: {1: 10.5, 2: 0.5},
+        2: {1: 5.5, 2: 10.5},
+        3: {1: 0.5, 2: 0.5},
+    }
+
+
+@pytest.mark.unit
+@pytest.mark.skipif(
+    not sklearn_avail, reason="sklearn (optional dependency) not available"
+)
+def test_optimal_num_clusters_elbow(dummy_data):
+    """Tests the get_optimal_num_clusters function"""
+    samples = generate_daily_data(dummy_data, horizon_length=2)
+    with pytest.raises(
+        NotImplementedError,
+        match=(
+            "elbow method is not supported currently for finding the optimal "
+            "number of clusters."
+        ),
+    ):
+        get_optimal_num_clusters(
+            samples=samples,
+            kmin=2,
+            kmax=7,
+            method="elbow",
+            generate_elbow_plot=True,
+            seed=20,
+        )
+
+
+@pytest.mark.unit
+@pytest.mark.skipif(
+    not sklearn_avail, reason="sklearn (optional dependency) not available"
+)
+def test_optimal_num_clusters_silhouette(dummy_data):
+    """Tests the get_optimal_num_clusters function"""
+    samples = generate_daily_data(dummy_data, horizon_length=2)
+    n_clusters = get_optimal_num_clusters(
+        samples=samples,
+        kmin=2,
+        kmax=7,
+        method="silhouette",
+        generate_elbow_plot=True,
+    )
+
+    assert n_clusters == 3
+
+    # Test that a figure was created
+    assert plt.gcf() is not None
+    # Test that axes were created
+    assert plt.gca() is not None
+    # Test that the plot has data
+    # assert plt.gca().has_data()
```

Comment:
Any reason why this test is commented out?

Reply:
It was a simple plot. I could see the data on the plot, but the assertion was failing for some unknown reason.
There isn't anything in the code that breaks it. So, I thought it would just be sufficient to run it for code coverage. 
I commented it for now. I will see if I can fix it later.